bcipy/language/tests/test_gpt2.py

```diff
@@ -0,0 +1,22 @@
+from bcipy.language.model.gpt2 import GPT2LanguageModel
+from bcipy.helpers.task import alphabet
+from bcipy.language.main import ResponseType
+
+
+if __name__ == "__main__":
+    symbol_set = alphabet()
+    response_type = ResponseType.SYMBOL
+    lm = GPT2LanguageModel(response_type, symbol_set, "gpt2")
+
+    next_char_pred = lm.state_update(list("does_it_make_sen"))
+    print(next_char_pred)
+    correct_char_rank = [c[0] for c in next_char_pred].index("S") + 1
+    print(correct_char_rank)
+    next_char_pred = lm.state_update(list("does_it_make_sens"))
+    print(next_char_pred)
+    correct_char_rank = [c[0] for c in next_char_pred].index("E") + 1
+    print(correct_char_rank)
+    next_char_pred = lm.state_update(list("does_it_make_sense"))
+    print(next_char_pred)
+    correct_char_rank = [c[0] for c in next_char_pred].index("_") + 1
+    print(correct_char_rank)
```

Comment:
Can you convert this into a test that will be discovered by pytest? This will allow the test to be run automatically (locally and on github during pull requests)

You can see examples of test structure in the other tests folders `bcipy/*/tests/`.
Here's an example for the EEG signal model: https://github.com/CAMBI-tech/BciPy/blob/1.5.1/bcipy/signal/tests/model/test_pca_rda_kde.py

You could make a class that inherits from `unittest.TestCase` - here's also an example structure to copy/paste from:
```python
import unittest

class TestNewThing(unittest.TestCase):
    def setUpClass(self):
        # if you want some heavyweight work to happen only once
        self.my_costly_attr = ...

    def setUp(self):
        # if you want some lightweight work to happen fresh before each test
        self.my_quick_attr = ...
        
    def test_first_feature(self):
        expected = ...
        observed = ...
        self.assertEqual(expected, observed)  

    def test_error_on_bad_action(self):
         with self.assertRaises(ValueError):
             self.model.do_something_invalid()
```

Reply:
If loading the model is too slow for a unit test (more than like ~1-10s), you can use the `@pytest.mark.slow` decorator on a function, or on the whole class, like this:
```python
@pytest.mark.slow
class MyTestCase(unittest.TestCase)
    @pytest.mark.slow
    def test_something_slow(self):
        ...
```

Then, people can run only the fast tests with `pytest -k "not slow"`, only the slow tests with `pytest -k "slow"`, etc

Slow tests are not (currently) run automatically during pull requests.
However, if the slow part is only downloading, we could still run this test during pull requests, and cache the downloaded file: https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows.
Setting that up would certainly take a bit of trial-and-error - happy to help with this.

Reply:
I agree that it would be great to have some unit tests. My preference would be if the underlying gpt2 model was mocked. We primarily want to test that your class is using it correctly. For instance, there are are a number of things you are managing in the `predict` method  (behavior changes depending on whether you are at the beginning of the word, etc). It would be good to demonstrate these conditions in a test. You may also want to have checks for your probability distributions.

Reply:
I agree that unit tests would be helpful. I can look into it but I prefer it in a follow-up (submit another PR for unit tests). Hope it's OK.

Reply:
Yes, please create a pivotal ticket for us to point during a refinement! https://www.pivotaltracker.com/n/projects/2460065